### DISCUSSION TITLE
To include/not include a year factor in models with spatiotemporal random effects

### DISCUSSION THREAD

**Discussion by larogers123** [sdmTMB/sdmTMB#232] [Q&A]
Is it necessary or advised to include a year factor in a model that has spatiotemporal random effects estimated? In my specific case, there is clearly a change in intercept (mean) by year. If I don't include a year fixed effect, the interannual change in mean gets sucked into the spatiotemporal random effects, such that they don't have a mean of zero within year. Is this a problem? The model predictions are very similar with and without the year factor (although AIC supports including year). 

I would default to including a year factor in my model(s), except that I'm also wanting to estimate an effect of annually-averaged temperature, T_y, and the model is overparameterized with an additional year factor. I'm aiming to draw inferences about that temperature effect, and not aiming to estimate an annual index per se. But again, interannual variation in the mean that is not explained by temperature appears to get taken up in the spatiotemporal term. I'm wondering if that is problematic. 

Thanks for any advice,
Lauren

**Comment by Lewis-Barnett-NOAA** [comment]
Hi Lauren,

It is not necessary.  I think the model structure you described can make sense.  Are you including any temporal structure on the spatiotemporal fields?

**Reply by larogers123** [reply]
The spatiotemporal fields are IID.

**Reply by Lewis-Barnett-NOAA** [reply]
Great, yeah I think that is the sensible choice in this context
